examples/social/src/contract.rs

```diff
@@ -91,56 +108,32 @@ impl Contract for SocialContract {
 }
 
 impl SocialContract {
-    async fn execute_post_operation(
-        &mut self,
-        text: String,
-        image_url: Option<String>,
-    ) -> (Destination, Message) {
+    async fn execute_post_operation(&mut self, text: String, image_url: Option<String>) {
         let timestamp = self.runtime.system_time();
         let post = OwnPost {
             timestamp,
             text,
             image_url,
         };
-        let index = self.runtime.emit(POSTS_CHANNEL_NAME.into(), &post);
+        let index = self.state.own_posts.count().try_into().unwrap();
         self.state.own_posts.push(post.clone());
-        (
-            ChannelName::from(POSTS_CHANNEL_NAME.to_vec()).into(),
-            Message::Post { index, post },
-        )
+        self.runtime
+            .emit(STREAM_NAME.into(), &Event::Post { post, index });
```

Comment:
I would say that `Post` shouldn't need an index and the events should be indexed internally - but that's orthogonal.

Reply:
The other chains need a way to refer to this specific post. I originally had used the event index here, but that's a bit wasteful: Some events aren't new posts.
(But the whole contract will need significant changes for https://github.com/linera-io/linera-protocol/issues/3782 anyway.)